command/v3/v3_unshare_service_command.go

```diff
@@ -0,0 +1,123 @@
+}
+
+//go:generate counterfeiter . ServiceInstanceSharedToActorV2
+
+type ServiceInstanceSharedToActorV2 interface {
+	GetSharedToSpaceGUID(serviceInstanceName string, sourceSpaceGUID string, sharedToOrgName string, sharedToSpaceName string) (string, v2action.Warnings, error)
+}
+
+type V3UnshareServiceCommand struct {
+	RequiredArgs    flag.ServiceInstance `positional-args:"yes"`
+	OrgName         string               `short:"o" required:"false" description:"Org of the other space (Default: targeted org)"`
+	SpaceName       string               `short:"s" required:"true" description:"Space to unshare the service instance from"`
+	usage           interface{}          `usage:"cf v3-unshare-service SERVICE_INSTANCE -s OTHER_SPACE [-o OTHER_ORG]"`
+	relatedCommands interface{}          `related_commands:"bind-service, service, services"`
+
+	UI          command.UI
+	Config      command.Config
+	SharedActor command.SharedActor
+	Actor       UnshareServiceActor
+	ActorV2     ServiceInstanceSharedToActorV2
+}
+
+func (cmd *V3UnshareServiceCommand) Setup(config command.Config, ui command.UI) error {
+	cmd.UI = ui
+	cmd.Config = config
+	cmd.SharedActor = sharedaction.NewActor(config)
+
+	client, _, err := shared.NewClients(config, ui, true)
+	if err != nil {
+		if v3Err, ok := err.(ccerror.V3UnexpectedResponseError); ok && v3Err.ResponseCode == http.StatusNotFound {
+			return translatableerror.MinimumAPIVersionNotMetError{MinimumVersion: ccversion.MinVersionShareServiceV3}
+		}
+		return err
+	}
+	cmd.Actor = v3action.NewActor(client, config, nil, nil)
+
+	ccClientV2, uaaClientV2, err := sharedV2.NewClients(config, ui, true)
+	if err != nil {
+		return err
+	}
+	cmd.ActorV2 = v2action.NewActor(ccClientV2, uaaClientV2, config)
+
+	return nil
+}
+
+func (cmd V3UnshareServiceCommand) Execute(args []string) error {
+	err := command.MinimumAPIVersionCheck(cmd.Actor.CloudControllerAPIVersion(), ccversion.MinVersionShareServiceV3)
+	if err != nil {
+		return err
+	}
+
+	cmd.UI.DisplayText(command.ExperimentalWarning)
+	cmd.UI.DisplayNewline()
+
+	err = cmd.SharedActor.CheckTarget(true, true)
+	if err != nil {
+		return err
+	}
+
+	user, err := cmd.Config.CurrentUser()
+	if err != nil {
+		return err
+	}
+
+	orgName := cmd.Config.TargetedOrganization().Name
+
+	if cmd.OrgName != "" {
+		orgName = cmd.OrgName
+	}
+
+	cmd.UI.DisplayTextWithFlavor("Unsharing service instance {{.ServiceInstanceName}} from org {{.OrgName}} / space {{.SpaceName}} as {{.Username}}...", map[string]interface{}{
+		"ServiceInstanceName": cmd.RequiredArgs.ServiceInstance,
+		"OrgName":             orgName,
+		"SpaceName":           cmd.SpaceName,
+		"Username":            user.Name,
+	})
+
+	sharedToSpaceGUID, warningsV2, err := cmd.ActorV2.GetSharedToSpaceGUID(cmd.RequiredArgs.ServiceInstance, cmd.Config.TargetedSpace().GUID, orgName, cmd.SpaceName)
+	cmd.UI.DisplayWarnings(warningsV2)
+
+	if err != nil {
+		if _, ok := err.(actionerror.ServiceInstanceNotSharedToSpaceError); ok {
```

Comment:
please handle the error conversion here: https://github.com/cloudfoundry/cli/blob/5923b92489b9dcb96bbc3c8ebb413fd27a1e18bb/command/translatableerror/convert_to_translatable_error.go#L17
and add a unit test here: https://github.com/cloudfoundry/cli/blob/5923b92489b9dcb96bbc3c8ebb413fd27a1e18bb/command/translatableerror/convert_to_translatable_error_test.go#L26

Reply:
Thanks, changes this to use ConvertToTranslatableErrror instead.